// Copyright 2025 Synnax Labs, Inc.
//
// Use of this software is governed by the Business Source License included in the file
// licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with the Business Source
// License, use of this software will be governed by the Apache License, Version 2.0,
// included in the file licenses/APL.txt.

syntax = "proto3";

package grpc.ranger;

import "core/pkg/api/ranger/pb/range.proto";
import "google/protobuf/empty.proto";

option go_package = "github.com/synnaxlabs/synnax/pkg/api/grpc/ranger";

service RangeCreateService {
  rpc Exec(CreateRequest) returns (CreateResponse) {}
}

service RangeRetrieveService {
  rpc Exec(RetrieveRequest) returns (RetrieveResponse) {}
}

service RangeDeleteService {
  rpc Exec(DeleteRequest) returns (google.protobuf.Empty) {}
}

service RangeRenameService {
  rpc Exec(RenameRequest) returns (google.protobuf.Empty) {}
}

message DeleteRequest {
  repeated string keys = 1;
}

message CreateRequest {
  repeated api.range.pb.Range ranges = 1;
}

message CreateResponse {
  repeated api.range.pb.Range ranges = 1;
}

message RetrieveRequest {
  repeated string keys = 1;
  repeated string names = 2;
}

message RetrieveResponse {
  repeated api.range.pb.Range ranges = 1;
}

message RenameRequest {
  string key = 1;
  string name = 2;
}
